openeo/rest/datacube.py

```diff
@@ -441,10 +441,23 @@ def load_stac(
         graph = PGNode("load_stac", arguments=arguments)
         try:
             metadata = metadata_from_stac(url)
+            # TODO: also apply spatial/temporal filters to metadata?
+
             if isinstance(bands, list):
-                # TODO: also apply spatial/temporal filters to metadata?
-                metadata = metadata.filter_bands(band_names=bands)
-        except Exception:
+                if not metadata.has_band_dimension():
+                    metadata = metadata.add_dimension(
+                        name="bands",
+                        type="bands",
+                        label=None,
+                    )
+                if all(metadata.band_dimension.contains_band(b) for b in bands):
+                    metadata = metadata.filter_bands(band_names=bands)
+                else:
+                    logging.warning(
+                        "Some bands are not available in the collection metadata. Using requested bands as is."
```

Comment:
The warning should list which provided bands bands are not consistent with bands listed in metadata, to make this warning more user friendly/actionable